[doc] Release notes new features for 2024.2 (#25005)
yugabyte/yugabyte-db: docs/content/preview/releases/yba-releases/v2024.2.md

**New Automatic On-premises Provisioning** {{<tags/feature/ga>}}

We've automated and simplified the previously manual or otherwise cumbersome process for preparing nodes for use in on-premises providers. System admins now download and run a single, simplified provisioning script directly on the VMs. The script not only provisions the node but also creates the on-premises infrastructure provider and adds the node to the infrastructure provider's free pool. After the process is completed, your nodes are ready for deployment in universes. With this new provisioning method, dozens of steps are reduced to just a few, the provisioning process won't change from release to release, and security concerns about providing SSH and/or sudo access to VMs are eliminated.
For more information, refer to [Add nodes to the on-premises provider](/stable/yugabyte-platform/configure-yugabyte-platform/on-premises-nodes/).

### New features

> * [xCluster Disaster Recovery](/stable/yugabyte-platform/back-up-restore-universes/disaster-recovery/). {{<tags/feature/ga>}} A turnkey solution for business continuity and disaster recovery, allowing you to recover from an unplanned outage (failover) or to perform a planned switchover. It also includes APIs to set up xCluster replication for a entire database.

Reviewer: ```suggestion
* [xCluster Disaster Recovery](/stable/yugabyte-platform/back-up-restore-universes/disaster-recovery/). A turnkey solution for business continuity and disaster recovery, allowing you to recover from an unplanned outage (failover) or to perform a planned switchover. It also includes APIs to set up xCluster replication for a entire database. {{<tags/feature/ga>}}
```

* [Semi-automatic xCluster replication](/stable/deploy/multi-dc/async-replication/async-replication-transactional/). Provides simplified management of YSQL transactional xCluster replication by operating at the database level instead of the table level. This reduces the need for IT admin involvement when tables are created or dropped, as xCluster management is only required for adding or removing entire databases from replication. DDL operations can now be performed by authorized users or DBAs without needing elevated privileges. {{<tags/feature/ea>}}

> * [Database registration simplification](/stable/yugabyte-platform/manage-deployments/ybdb-releases/). Simplified process for registering database versions (and their binaries) with YugabyteDB Anywhere using a dramatically improved user interface.

Reviewer: ```suggestion
* [Simplified database registration](/stable/yugabyte-platform/manage-deployments/ybdb-releases/). Simplified process for registering database versions (and their binaries) with YugabyteDB Anywhere using a dramatically improved user interface.
```

* Force update of TLS certificates. If certificates expire, you can now "force update" the replacement of certificates with new ones, removing the need to call Technical Support.

> * [Clock sync alert](/stable/yugabyte-platform/prepare/server-nodes-software/). Fire a clock sync alert if the Network Time Protocol Daemon (ntpd) is not enabled or not running on a node of a YugabyteDB universe. {{<tags/feature/ea>}}

Reviewer: ```suggestion
* [Clock sync alert](/stable/yugabyte-platform/alerts-monitoring/alert/). Fire a clock sync alert if the Network Time Protocol Daemon (ntpd) is not enabled or not running on a node of a YugabyteDB universe. {{<tags/feature/ea>}}
```

* [Backup portability](/stable/yugabyte-platform/back-up-restore-universes/restore-universe-data/#advanced-restore-procedure). After sending backups to a cloud object store (such as S3, GCS, or Azure blob storage), you can now copy those backup sets to other cloud storage buckets and restore from them.

* Kubernetes
  * [Kubernetes Resource Specification](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#instance-configuration). You can now configure CPU and memory needs for YB-TServer and YB-Master pods natively, instead of using the instance types API.

>   * [K8s service endpoint documentation](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#create-common-yb-tserver-service-for-zones). Documented the workflow to create a unified service endpoint for all TServers in single namespace universes.

Reviewer: ```suggestion
  * [Kubernetes service endpoint documentation](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#create-common-yb-tserver-service-for-zones). Documented the workflow to create a unified service endpoint for all TServers in single namespace universes.
```

  * [Kubernetes readiness probe](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#configure-helm-overrides). Added readiness probes for TServer and Master pods in YugabyteDB, supporting custom or default configurations. This enhances stability by ensuring YSQL/YCQL and YB-Master pods are ready before traffic is routed.

>   * [K8s service endpoint](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#create-a-common-load-balancer-service-for-yb-masters-yb-tservers). For single-region universes in Kubernetes, added support for creating a unified service endpoint. This is in addition to the already-supported one service endpoint per availability zone (AZ). {{<tags/feature/ea>}}

Reviewer: ```suggestion
  * [Kubernetes service endpoint](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#create-a-common-load-balancer-service-for-yb-masters-yb-tservers). For single-region universes in Kubernetes, added support for creating a unified service endpoint. This is in addition to the already-supported one service endpoint per availability zone (AZ). {{<tags/feature/ea>}}
```

  * [ARM processor support for Kubernetes universes](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/). Added support for running Kubernetes-based universes on ARM CPUs. {{<tags/feature/tp>}}

>   * Non-disruptive flag changes for K8s. Adding and modifying selected configuration flags for YB-Master and YB-TServer nodes in a Kubernetes universe can now be done without restarting the Master or T-Server processes.

Reviewer: ```suggestion
  * Non-disruptive flag changes for K8s. Adding and modifying selected configuration flags for YB-Master and YB-TServer pods in a Kubernetes universe can now be done without restarting the Master or T-Server processes.
```

* [YBA HA enhancements](/stable/yugabyte-platform/troubleshoot/universe-issues/#run-pre-checks-before-edit-and-upgrade-operations). Added validation checks to ensure split-brain correctness post failover. Specifically, after a failover to the standby YugabyteDB Anywhere instance, if the new primary YugabyteDB Anywhere detects that its state is out-of-sync with a universe it's operating on, it aborts the operation.

* [Support bundle collection improvements](/stable/yugabyte-platform/troubleshoot/universe-issues/#use-support-bundles). Support bundles now optionally include time series metrics. Also, collection is more resilient to point failures; if one data collection task fails, the collection operation ignores the failure and proceeds.

> * [Connection pooling](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone/#advanced-configuration). Optional server-side Connection Pooling enables more connections per cluster, more efficient memory usage, and faster first connection (and therefore first query) latencies. {{<tags/feature/tp>}}

Reviewer: ```suggestion
* [Connection pooling](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone/#advanced-configuration). Optional server-side connection pooling enables more connections per cluster, more efficient memory usage, and faster first connection (and therefore first query) latencies. {{<tags/feature/tp>}}
```

* [Improved node and process metrics](/stable/yugabyte-platform/alerts-monitoring/anywhere-metrics/#per-process). Metrics are now collected about each node's processes (PG, YBC, node agent, otel agent, and so on). Metrics cover up times, activity counts, cpu/memory consumed, and more.

> * [Non-root installation of YBA](/stable/yugabyte-platform/install-yugabyte-platform/install-software/installer/). You can install YugabyteDB Anywhere as a de-privileged user; a root-privileged user is no longer required.  This reduces friction and security concerns during initial testing, setup, and proof of concepts. {{<tags/feature/ea>}}

Reviewer: ```suggestion
* [Non-root installation of YBA](/stable/yugabyte-platform/install-yugabyte-platform/install-software/installer/#non-sudo-installation). You can install YugabyteDB Anywhere as a de-privileged user; a root-privileged user is no longer required.  This reduces friction and security concerns during initial testing, setup, and proof of concepts. {{<tags/feature/ea>}}
```

* [Backup portability](/stable/yugabyte-platform/back-up-restore-universes/restore-universe-data/#advanced-restore-procedure). After sending backups to a cloud object store (such as S3, GCS, or Azure blob storage), you can now copy those backup sets to other cloud storage buckets and restore from them.

* Kubernetes

>   * [Kubernetes resource Specification](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#instance-configuration). You can now configure CPU and memory needs for YB-TServer and YB-Master pods natively, instead of using the instance types API.

Reviewer: ```suggestion
  * [Kubernetes resource specification](/stable/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#instance-configuration). You can now configure CPU and memory needs for YB-TServer and YB-Master pods natively, instead of using the instance types API.
```
